cmd/atlas/internal/client/driver.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"ariga.io/atlas/sql/mysql"
+	"ariga.io/atlas/sql/postgres"
+	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlite"
+)
+
+type (
+	// Driver provides to diff, execute and inspect against a Data Source.
+	Driver interface {
+		schema.Differ
+		schema.Execer
+		schema.Inspector
+	}
+	// AtlasDriver implements the Driver interface using Atlas.
+	AtlasDriver struct {
```

Comment:
consider what will happen when you have more subcommand in the client, this struct will become huge and have a lot of dependencies. dependencies lead to coupling, and coupling leads to complexity. (complexity leads to the dark side, said master yoda)

It's better to have many specialized types for each group of operations. i think it would make sense to have a struct for `Schema` related commands (currently inspect and apply). later when we add other domains of functionality, add more specific structs.